hera_sim/simulate.py

```diff
@@ -754,64 +761,29 @@ def _get_model_parameters(model):
         return model_params
 
     @staticmethod
-    def _get_component(component):
-        # TODO: docstring
-        """
-        """
-        try:
-            if issubclass(component, SimulationComponent):
-                # support passing user-defined classes that inherit from
-                # the SimulationComponent base class to add method
-                return component, True
-            else:
-                # issubclass will not raise a TypeError in python <= 3.6
-                raise TypeError
-        except TypeError:
-            # this is raised if ``component`` is not a class
-            if component.__class__.__name__ == "function":
-                raise TypeError(
-                    "You are attempting to add a component that is "
-                    "modeled using a function. Please convert the "
-                    "function to a callable class and try again."
-                )
-            if callable(component):
-                # if it's callable, then it's either a user-defined
-                # function or a class instance
-                return component, False
-            if not isinstance(component, str):
-                # TODO: update this error message to reflect the
-                # change in allowed component types
-                raise TypeError(
-                    "``component`` must be either a class which "
-                    "derives from ``SimulationComponent`` or an "
-                    "instance of a callable class, or a function, "
-                    "whose signature is:\n"
-                    "func(lsts, freqs, *args, **kwargs)\n"
-                    "If it is none of the above, then it must be "
-                    "a string which corresponds to the name of a "
-                    "``hera_sim`` class or an alias thereof."
+    def _get_component(
+        component: [str, Type[SimulationComponent], SimulationComponent]
+    ) -> Tuple[Union[SimulationComponent, Type[SimulationComponent]], bool]:
+        """Given an input component, normalize the output to be either a class or instance.
+
+        """
+        if np.issubclass_(component, SimulationComponent):
+            return component, True
+        elif isinstance(component, str):
+            try:
+                return get_model(component), True
+            except KeyError:
+                raise ValueError(
+                    f"The model '{component}' does not exist. The following models are "
+                    f"available: \n{print_all_components()}."
```

Comment:
Using `print_all_components()` here returns `None`, and so the output of this error message isn't what's expected. We've got (at least) two options: make `print_all_components` return the string literal, or turn `print_all_components` into a thin wrapper for a function `list_all_components` that does essentially what `print_all_components` currently does, but returns the string literal rather than printing it.

Reply:
Good catch. I chose to rename the function to list_all_components and update it to return the string. I think printing inside the function is kinda silly.